internal/pkg/addon/addons.go

```diff
@@ -86,17 +93,98 @@ func (a *Addons) Template() (string, error) {
 	return string(out), nil
 }
 
-func filterYAMLfiles(files []string) []string {
-	yamlExtensions := []string{".yaml", ".yml"}
+// Parameters returns the content of user-defined additional CloudFormation Parameters
+// to pass from the parent stack to Template.
+//
+// If there is no addons/ directory defined, then returns "" and ErrAddonsNotFound.
+// If there are addons but no parameters file defined, then returns "" and nil for error.
+// If there are multiple parameters files, then returns "" and cannot define multiple parameter files error.
+// If the addons parameters use the reserved parameter names, then returns "" and a reserved parameter error.
+func (a *Addons) Parameters() (string, error) {
+	fnames, err := a.ws.ReadAddonsDir(a.wlName)
+	if err != nil {
+		return "", &ErrAddonsNotFound{
+			WlName:    a.wlName,
+			ParentErr: err,
+		}
+	}
+	paramFiles := filterFiles(fnames, paramsMatcher)
+	if len(paramFiles) == 0 {
+		return "", nil
+	}
+	if len(paramFiles) > 1 {
+		return "", fmt.Errorf("defining multiple %s is not allowed under %s addons", english.WordSeries(parameterFileNames, "and"), a.wlName)
+	}
+	raw, err := a.ws.ReadAddon(a.wlName, paramFiles[0])
+	if err != nil {
+		return "", fmt.Errorf("read parameter file %s under %s addons: %w", paramFiles[0], a.wlName, err)
+	}
+	content := struct {
+		Parameters yaml.Node `yaml:"Parameters"`
+	}{}
+	if err := yaml.Unmarshal(raw, &content); err != nil {
+		return "", fmt.Errorf("unmarshal 'Parameters' in file %s under %s addons: %w", paramFiles[0], a.wlName, err)
+	}
+	if content.Parameters.IsZero() {
+		return "", fmt.Errorf("must define field 'Parameters' in file %s under %s addons", paramFiles[0], a.wlName)
+	}
+	if err := a.validateReservedParameters(content.Parameters, paramFiles[0]); err != nil {
+		return "", err
+	}
+	buf := new(strings.Builder)
+	encoder := yaml.NewEncoder(buf)
+	encoder.SetIndent(2 /* 2 spaces to indent */)
+	if err := encoder.Encode(content.Parameters); err != nil {
+		return "", fmt.Errorf("marshal contents of 'Parameters' in file %s under %s addons", paramFiles[0], a.wlName)
+	}
+	return buf.String(), nil
+}
+
+func (a *Addons) validateReservedParameters(params yaml.Node, fname string) error {
+	content := struct {
+		App  yaml.Node `yaml:"App"`
+		Env  yaml.Node `yaml:"Env"`
+		Name yaml.Node `yaml:"Name"`
+	}{}
+	if err := params.Decode(&content); err != nil {
+		return fmt.Errorf("decode content of parameters file %s under %s addons", fname, a.wlName)
+	}
+
+	for _, field := range []yaml.Node{content.App, content.Env, content.Name} {
+		if !field.IsZero() {
+			return fmt.Errorf("reserved parameters 'App', 'Env', and 'Name' cannot be declared in %s under %s addons", fname, a.wlName)
```

Comment:
Just a callout but when in future we add more reserved parameters it could be a breaking change for some users.

Reply:
We should be safe 💭 because we can't add more reserved parameters otherwise it will break existing users addon templates